odoo/addons/base/transpiler/transpiler_js.py

```diff
@@ -0,0 +1,393 @@
+                \n'use strict';\
+                \nlet __exports = {{}};\
+                \n{self.content}\
+                \nreturn __exports;\
+                \n}});\n"
+
+    def convert_function_and_class_export(self, default=False):
+        """
+        Transpile functions and classes that are being exported.
+
+        before: export function name
+        after: const name = __exports.name = function name
+
+        before: export async function name
+        after: const name = __exports.name = async function name
+
+        before: export class name
+        after: const name = __exports.name = class name
+
+        if default:
+        before: export default function name
+        after: const name = __exports[Symbol.for("default")] = function name
+        (same for the rest)
+
+        :param default: should we convert the default exports as well ?
+        """
+        pattern = r"^(?P<space>\s*)export\s+(?P<type>(async\s+)?function|class)\s+(?P<identifier>\w+)"
+        repl = r"\g<space>const \g<identifier> = __exports.\g<identifier> = \g<type> \g<identifier>"
+        if default:
+            pattern = r"^(?P<space>\s*)export\s+default\s+(?P<type>(async\s+)?function|class)\s+(?P<identifier>\w+)"
+            repl = r"""\g<space>const \g<identifier> = __exports[Symbol.for("default")] = \g<type> \g<identifier>"""
+        p = re.compile(pattern, re.MULTILINE)
+        self.content = p.sub(repl, self.content)
+
+    def convert_variable_export(self, default=False):
+        """
+        Transpile variables that are being exported.
+
+        before: export let name =
+        after: let name = __exports.name =
+        (same with var and const)
+
+        if default:
+        before: export let name =
+        after: let name = __exports[Symbol.for("default")] =
+
+        :param default: should we convert the default exports as well ?
+        """
+        pattern = r"^(?P<space>\s*)export\s+(?P<type>let|const|var)\s+(?P<identifier>\w+)\s*="
+        repl = r"\g<space>\g<type> \g<identifier> = __exports.\g<identifier> ="
+        if default:
+            pattern = r"^(?P<space>\s*)export\s+default\s+(?P<type>let|const|var)\s+(?P<identifier>\w+)\s*="
+            repl = r"""\g<space>\g<type> \g<identifier> = __exports[Symbol.for("default")] ="""
+        p = re.compile(pattern, re.MULTILINE)
+        self.content = p.sub(repl, self.content)
+
+    def convert_list_export(self):
+        """
+        Transpile exports of multiple elements
+
+        before: export { a, b, c as x }
+        after: Object.assign(__exports, { a, b, x: c })
+        """
+        p = re.compile(r"^(?P<space>\s*)export\s*(?P<list>{(\s*\w+\s*,?\s*)*}\s*)", re.MULTILINE)
+
+        def repl(matchobj):
+            d = matchobj.groupdict()
+            list_process = "{" + ", ".join([convert_as(val) for val in d.get("list")[1:-1].split(",")]) + "}"
+            space = d.get("space", "")
```

Comment:
`groupdict` is redundant, can just index the match object.

And `.get` will never do anything useful for regex: all groups are always present in the output, groups which were not matched (if there is an alternation) are mapped to a `None`, so the default value can never be used.